src/puptoo/process/profile.py

```diff
@@ -406,19 +410,25 @@ def system_profile(
             catch_error("uname", e)
             raise
 
-    if redhat_release:
+    if redhat_release and os_release:
         try:
-            profile["os_release"] = redhat_release.rhel
-            profile["operating_system"] = {
-                "major": redhat_release.major,
-                "minor": redhat_release.minor,
-                "name": "RHEL",
-            }
-            if profile.get("host_type") is None:
-                if redhat_release.major >= 8:
-                    profile["system_update_method"] = "dnf"
-                else:
-                    profile["system_update_method"] = "yum" 
+            if os_release.release == "RHEL" or os_release.name == "CentOS Linux":
+                if redhat_release.minor is None:
+                    redhat_release.minor = 0
```

Comment:
It's not good to write back to the `redhat_release` which is a `shared` component (parser).

Better introduce a local variable for it, e.g. `minor = redhat_release.minor or 0`

Other than this, the PR looks good to me.